query/src/main/java/org/infinispan/query/clustered/commandworkers/CQCreateEagerQuery.java

```diff
@@ -56,7 +56,7 @@ private CompletionStage<NodeTopDocs> collectKeys(SearchQueryBuilder query) {
                long hitCount = queryResult.total().hitCount();
 
                Object[] keys = queryResult.hits().stream()
-                     .map(EntityReference::key)
+                     .map(EntityReference::id)
```

Comment:
This is correct, but if you're only going to use the `id`, you might want to use the [`id` projection](https://docs.jboss.org/hibernate/search/6.2/reference/en-US/html_single/#search-dsl-projection-id) instead of the `entityReference` projection.

Reply:
good idea. thank you. going to apply it

Reply:
@yrodiere trying to apply it I noticed that while `EntityReference::id` returns the cache entry key, correctly applying the `KeyTransformationHandlerIdentifierBridge`, returning for instance `BigGoat`. In case of id projection the bridge is not applied so we get the raw Lucene document identifier, returning for instance `S:BigGoat`.

Reply:
I don't know if it is possible to ask for a id projection that applies the `IdentifierBridge`. Maybe we should that manually?

Reply:
The ID projection definitely applies the bridge, otherwise most of our tests would be failing since we'd be getting a `String` everywhere. Might have something to do with how Infinispan transforms entity references...

Reply:
checking...

Reply:
I'll check whether, maybe, provided identifiers (which only Infinispan uses) behave differently

Reply:
It's most definitely a bug for provided identifiers, I'll look into it.

Reply:
See https://hibernate.atlassian.net/browse/HSEARCH-4877, https://github.com/hibernate/hibernate-search/pull/3539